scripts/backfill_source_tags_for_unregistered_contributors.py

```diff
@@ -0,0 +1,167 @@
+from datetime import timedelta
+import logging
+from django.apps import apps
+from website.app import init_app
+from website.util.metrics import ProviderSourceTags, ProviderClaimedTags, CampaignSourceTags, CampaignClaimedTags, provider_source_tag, provider_claimed_tag
+from scripts import utils as script_utils
+import datetime
+from django.core.paginator import Paginator
+from django.db.models import F
+from tqdm import tqdm
+
+import sys
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+def _get_set_of_user_ids_from_logs(node_logs):
```

Comment:
I can probably eliminate these python loops too. But will put the current script to test and see if the previous optimizations work and whether we need more.